initial commit for tidymodels 2023 q1 post (#637)
tidyverse/tidyverse.org: content/blog/tidymodels-2023-q1/index.Rmd

-   [Tuning hyperparameters with tidymodels is a delight](https://www.tidyverse.org/blog/2023/04/tuning-delights/)
-   [censored 0.2.0](https://www.tidyverse.org/blog/2023/04/censored-0-2-0/)
-   [The tidymodels is getting a whole lot faster](https://www.simonpcouch.com/blog/speedups-2023/)

Since [our last roundup post](https://www.tidyverse.org/blog/2022/12/tidymodels-2022-q4/), there have been CRAN releases of `r nrow(tm_data)` tidymodels packages. Here are links to their NEWS files:

```{r}
#| echo: FALSE
#| results: asis

cat(tm_data$news, sep = "\n")
```

We'll highlight a few especially notable changes below: more informative errors and faster code. First, loading the collection of packages:

```{r}
#| eval: FALSE
library(tidymodels)
library(embed)

data("ames", package = "modeldata")
```

## More informative errors

> We like to make sure that the functions do what they are supposed to do. In the last months we are focused on making it so functions throw errors in such a way that they are easier for the users to pinpoint what went wrong and where. Since the modeling pipeline can be quite complicated, getting uninformative errors are a no-go.

Reviewer: "have been focused" :)

> We like to make sure that the functions do what they are supposed to do. In the last months we are focused on making it so functions throw errors in such a way that they are easier for the users to pinpoint what went wrong and where. Since the modeling pipeline can be quite complicated, getting uninformative errors are a no-go.

Reviewer 2: Edited my suggestion, thanks🥑

Across the tidymodels, error messages will now indicate the user-facing function that caused the error rather than the internal function that it came from.

From dials, an error that looked like

```r
degree(range = c(1L, 5L))
#> Error in `new_quant_param()`:
#> ! Since `type = 'double'`, please use that data type for the range.
```

Now says that the error came from `degree()` rather than `new_quant_param()`

```{r}
#| error: TRUE
degree(range = c(1L, 5L))
```

The same thing can be seen with the yardstick metrics

``` r
mtcars |>
  accuracy(vs, am)
#> Error in `dplyr::summarise()`:
#> ℹ In argument: `.estimate = metric_fn(truth = vs, estimate = am, na_rm =
#>   na_rm)`.
#> Caused by error in `validate_class()`:
#> ! `truth` should be a factor but a numeric was supplied.
```

> which now errors much better

Reviewer 2: ```suggestion
which now errors much more informatively
```

```{r}
#| error: TRUE
mtcars |>
  accuracy(vs, am)

> ```

Reviewer 2: Love that you included both of these examples. Such a nice contrast. :)

> Lastly, one of the biggest improvements came in recipes, which now shows which step caused the error instead of saying it happened in `prep()` or `bake()`. This is a huge improvement since you are likely to have many steps.

Reviewer 2: ```suggestion
Lastly, one of the biggest improvements came in recipes, which now shows which step caused the error instead of saying it happened in `prep()` or `bake()`. This is a huge improvement since preprocessing pipelines often which can string together many preprocessing steps.
```

Reviewer: "often which can string" -> "which often string"

Reviewer 2: Updated, thank you!

[Principal Component Analysis](https://en.wikipedia.org/wiki/Principal_component_analysis) is a really powerful and fast method for dimensionality reduction of large data sets. However, for data with many columns, it can be computationally expensive to calculate all the principal components. `step_pca_truncated()` works in much the same way as `step_pca()` but it only calculates the number of components it needs

```{r}
pca_normal <- recipe(Sale_Price ~ ., data = ames) |>
  step_dummy(all_nominal_predictors()) |>
  step_pca(all_numeric_predictors(), num_comp = 3)

pca_truncated <- recipe(Sale_Price ~ ., data = ames) |>
  step_dummy(all_nominal_predictors()) |>
  step_pca_truncated(all_numeric_predictors(), num_comp = 3)
```

```{r}
tictoc::tic()
prep(pca_normal) |> bake(ames)
tictoc::toc()
```

```{r}
tictoc::tic()
prep(pca_truncated) |> bake(ames)
tictoc::toc()
```

The speedup will be orders of magnitude larger for very wide data.

## Acknowledgements

We'd like to thank those in the community that contributed to tidymodels in the last quarter:

```{r}
#| echo: FALSE
#| results: asis

cat(tm_data$thanks, sep = "\n")
```

> We're grateful for all of the tidymodels community, from observers to users to contributors, and wish you all a happy new year!

Reviewer 2: ```suggestion
We're grateful for all of the tidymodels community, from observers to users to contributors. Happy modeling!
```

I belieeeeve the "happy new year" is an artifact from last quarter's post?

Author: it sure is! 😄
